### ISSUE TITLE
Create automated build for the executable

### ISSUE THREAD

**Issue by mrbiggred** [corgibytes/freshli-cli#7]
**Edit:** Changed this case to create an automated build for the CLI.  Didn't notice one did not exist yet.  Leaving the below notes on GitVersion as using GitVersion will be part of the build process to determine the exe version.

Use [GitVersion](https://github.com/GitTools/GitVersion) to generate the version number for this project.  Actually the plan is to use the [GitHub Action](https://github.com/marketplace/actions/gittools).  GitVersion will be setup to support the Cactus Branching model (see below for details) which requires a setup like:

```
# GitVersion.yml
branches:
  master:
    mode: ContinuousDeployment
    tag: 'alpha'
    increment: Minor
    prevent-increment-of-merged-branch-version: false
    track-merge-target: true
  release:
    tag: 'beta'
    increment: Patch
    prevent-increment-of-merged-branch-version: true
    track-merge-target: false
```

The idea is that only `release-X.Y` branches are tagged something like `vX.Y.Z`.  Then this tag is merged back to `main` GitVersion will calculate the next version number as `vX.Y+1.0.0`.

When a feature branch is merged to `release-X.Y` for a bug fix only the patch number will be incremented and the major, minor, and patch numbers left as is.

GitVersion generates several different version number formats such as:

- AssemblySemVer: 0.8.0.0
- NuGetVersionV2: 0.8.0-alpha0001
- MajorMinorPatch: 0.8.0

GitVersion can set the version in C# Assembly Version files now that they have fixed this [issue](https://github.com/GitTools/actions/issues/125).

----

An example of the Cactus or Trident Branching model:

![image](https://user-images.githubusercontent.com/17273852/109237356-9f809e80-778e-11eb-92e6-65ae1480e584.png)

**Comment by mrbiggred** [comment]
@mfcorgi I updated this issues title to clarify it's purpose.  This issue is for updating the version on the exe.  So when when exe is built you should be able to right click on it in Windows and see it has a version of vX.Y.Z.

Now that I look I don't see any GitHub workflows for this repo.  That means this case just became creating a basic build.  The build should compile do the usual items: compile, run tests (if any), along with versioning the executable.  You should be able to borrow from the Freshli Lib [dotnet-core.yml](https://github.com/corgibytes/freshli-lib/blob/main/.github/workflows/dotnet-core.yml) file.

**Comment by mfcorgi** [comment]
> @mfcorgi I updated this issues title to clarify it's purpose. This issue is for updating the version on the exe. So when when exe is built you should be able to right click on it in Windows and see it has a version of vX.Y.Z.
> 
> Now that I look I don't see any GitHub workflows for this repo. That means this case just became creating a basic build. The build should compile do the usual items: compile, run tests (if any), along with versioning the executable. You should be able to borrow from the Freshli Lib [dotnet-core.yml](https://github.com/corgibytes/freshli-lib/blob/main/.github/workflows/dotnet-core.yml) file.

I believe, if I am not wrong, that the difference would be that Freshli Lib uploads the artifact to nuget and in this case, I need to zip the Release folder content and upload it to github.

**Comment by mrbiggred** [comment]
Closed by PR #17.
